txcache/baseTxCache.go

```diff
@@ -0,0 +1,46 @@
+package txcache
+
+import (
+	"sync"
+
+	"github.com/multiversx/mx-chain-storage-go/common"
+)
+
+const maxNumOfEvictionWorkers = 5
+
+type evictionWorkerPool interface {
+	Stop()
+	Submit(task func())
+}
+
+type baseTxCache struct {
+	mutEvictionHandlers sync.RWMutex
+	evictionHandlers    []func(txHash []byte)
```

Comment:
can we have instead of function pointers an array of interfaces?
```go
type EvictionNotifier interface {
    NotifyEviction(txHash []byte)
    IsInterfaceNill() bool
}

type baseTxCache struct {
    mutEvictionHandlers sync.RWMutex
    evictionHandlers    []EvictionNotifier
}
```

Reply:
updated